### DISCUSSION TITLE
Using Nuxt Module from source without calling prepack before

### DISCUSSION THREAD

**Discussion by mklueh** [nuxt/nuxt#16747] [Questions]
### Describe the feature

For my understanding, a Nuxt module has to be prepacked first and Nuxt is working with the resulting .cjs / .mjs files.

In my case I'm having an NX workspace with one Nuxt 3 application and one Nuxt 3 module.
The required prepack step has the consequence, that I need to perform extra steps during my CI workflow to prepack one or more modules first before I call build.

Also during development it is restrictive, if you are used the hot-reloading of Nuxt, but when it comes to module development you have to stop your application, call prepack again and restart your application then, to test it.

Maybe it is already possible, according to the docs I assume it is not.

https://github.com/nuxt/framework/discussions/7791

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by cpreston321** [comment]
@mklueh https://github.com/nuxt/framework/discussions/7791#discussioncomment-3737523

**Comment by danielroe** [comment]
I would also add that you can very easily point a module to a TS source file also.

```js
export default defineNuxtConfig({
  modules: [
    '~/modules/some-module',
  ],
})
```

```js
// ~/modules/some-module.ts
import { defineNuxtModule, useNuxt } from '@nuxt/kit'

export default defineNuxtModule({
  meta: {
    name: 'some-module',
  },
  setup() {
    const nuxt = useNuxt()
    // ...
  },
})
```
